### PR TITLE
create network infra

### PR DESCRIPTION
**Pull request by delmar74**

### CODE REVIEW

File: network.tf [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+# VPC
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+}
+
+# subnet
+resource "aws_subnet" "public" {
+
+  count = 2
```
Comment (vladimirmukhin) [inline network.tf:14]:
```suggestion
resource "aws_subnet" "public" {
  count = 2
```

File: network.tf [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  vpc_id     = aws_vpc.main.id
+  cidr_block = local.public_cidr[count.index]
+
+  tags = {
+    Name = "public${count.index}"
+  }
+}
+
+resource "aws_subnet" "private" {
+  count = 2
```
Comment (vladimirmukhin) [inline network.tf:25]: Good job with using `count` here.

However, the task says `at least two subnets` this means that we might encounter a situation where 3 subnets per layer are required or even 5

You need to create more universal approach that allows setting `count` value dynamically:
https://www.terraform.io/docs/language/functions/length.html

File: network.tf [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  tags = {
+    Name = "main"
+  }
+}
+
+# EIP
+resource "aws_eip" "nat" {
+  count = 2
+
+  vpc      = true
```
Comment (vladimirmukhin) [inline network.tf:48]:
```suggestion
  vpc = true
```

File: network.tf [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+resource "aws_nat_gateway" "ngw" {
+  count = 2 
+
+  allocation_id = aws_eip.nat[count.index].id
+  subnet_id     = aws_subnet.public[count.index].id
+
+  tags = {
+    Name = "main"
+  }
+  #depends_on = [aws_internet_gateway.example]
```
Comment (vladimirmukhin) [inline network.tf:61]: It's completely up to your preference. However, having commented-out code is considered a bad styling

File: network.tf [hunk 5]
```diff
@@ -0,0 +1,91 @@
+locals {
+    public_cidr = ["10.0.0.0/24", "10.0.1.0/24"]
+    private_cidr = ["10.0.10.0/24", "10.0.11.0/24"]
```
Comment (vladimirmukhin) [inline network.tf:3]:
```suggestion
  public_cidr = ["10.0.0.0/24", "10.0.1.0/24"]
  private_cidr = ["10.0.10.0/24", "10.0.11.0/24"]
```

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: network.tf (hunk 1 above)
Comment (delmar74) [reply network.tf]: Ok

File: network.tf (hunk 4 above)
Comment (delmar74) [reply network.tf]: Clear

**Pull request merged**